src/main/java/echoquery/sql/Querier.java

```diff
@@ -21,11 +24,98 @@ public Querier(Connection conn) {
     this.inferrer = new SchemaInferrer(conn);
   }
 
-  public QueryResult execute(QueryRequest request) throws SQLException {
+  public QueryResult execute(QueryRequest request) {
+    // First validate the request, interactively mending malformed requests.
+    Optional<QueryResult> invalidation = validate(request);
```

Comment:
I think having a validator validating requests makes more sense than every request validating itself, more of a pipeline